src/compiler/preprocess/index.ts

```diff
@@ -109,6 +183,13 @@ function get_replacement(
 		if (typeof(decoded_map.mappings) === 'string') {
 			decoded_map.mappings = decode_mappings(decoded_map.mappings);
 		}
+		if (
```

Comment:
Do we need to check for the presence of `_mappings` or is the second half of the check sufficient?

If we need both I might tweak the spacing a bit. It's a bit uncommon to me to have nothing else on the same line as the `if`. For example:

```
		if ((decoded_map as any)._mappings &&
				decoded_map.constructor.name == 'SourceMapGenerator') {
```

Reply:
we check both for security, to reduce chances of a false positive
_mappings is checked first for performance, string compare is slower

my verbose if-syntax is prettier, cos all sub-expressions are on the same indent-level

Reply:
I think it'd all fit onto one line? it looks like it'd be just about 100 characters. Then we could avoid figuring out what indentation style to use :smile:  we could rename `decoded_map` to `decoded` to make it fit better. I think the `map` part would be obviously implied and not really necessary

Reply:
> I think it'd all fit onto one line?

done

97 chars per line with 2-space-per-tab indent
lets hope no one will read this in a 80 char terminal